bioblend/_tests/TestGalaxyWorkflows.py

```diff
@@ -205,15 +205,76 @@ def test_run_workflow(self):
             self.gi.workflows.run_workflow(wf['id'], None)
 
     def test_invoke_workflow(self):
+        invocation = self._invoke_workflow()
+        assert invocation['state'] == 'new', invocation
+
+    def test_create_workflow_from_history(self):
+        invocation = self._invoke_workflow()
+
+        time.sleep(5)
+        job = sorted(self.gi.jobs.get_jobs(), key=lambda x: x['create_time'], reverse=True)[0]
+
+        for _ in range(20):
+            [_job] = [_j for _j in self.gi.jobs.get_jobs() if _j['id'] == job['id']]
+            if _job['id'] == job['id'] and _job['state'] == 'ok':
+                break
+            time.sleep(.5)
+
+        wf1 = self.gi.workflows.show_workflow(invocation['workflow_id'])
+        datasets = self.gi.histories.show_history(invocation['history_id'], contents=True)
+        dataset_hids = [dataset['hid'] for dataset in datasets]
+        job_ids = [job['id']]
```

Comment:
What about something like that (not tested)? It would require annotating this test with `@test_util.skip_unless_galaxy('release_19.09')`:
```suggestion
        invocation = self.gi.invocations.wait_for_invocation(invocation['id'])
        wf1 = self.gi.workflows.show_workflow(invocation['workflow_id'])
        datasets = self.gi.histories.show_history(invocation['history_id'], contents=True)
        dataset_hids = [dataset['hid'] for dataset in datasets]
        job_ids = [step['job_id'] for step in invocation['steps]]
```

Reply:
Testing locally with `dev, py39` I get `bioblend.ConnectionError: Unexpected HTTP status code: 400: {"err_msg": "Malformed id ( None ) specified, unable to decode", "err_code": 400009, "id": "None"}` with `job_ids = [None, None, 'ebfb8f50c6abde6d']`

Reply:
We only want the last job, as the first two are dataset uploads. @rikeshi can you therefore try replacing the last line with `jobs_ids = invocation['steps'][-1]['job_id']`?

Reply:
`bioblend.ConnectionError: Unexpected HTTP status code: 500: {"err_msg": "Uncaught exception in exposed API method:", "err_code": 0}`

Reply:
I've update the suggestion above by adding a ` if step['job_id']` to the list comprehension that is assigned to `job_ids`. This should work and remain generic.

Reply:
I have applied the suggestion but it would not work without waiting for the job state. I did make the loop a bit cleaner. I am not aware of a better way to do this at the moment.